src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.Factory.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+
+#pragma warning disable CS8601 // Possible null reference assignment.
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+#pragma warning disable 8500 // address / sizeof of managed types
+
+namespace System.Numerics.Tensors
+{
+    public static partial class Tensor
+    {
+        public static Tensor<T> Create<T>(bool mustPin, ReadOnlySpan<nint> lengths)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanHelpers.CalculateTotalLength(lengths);
+            T[] values = mustPin ? GC.AllocateArray<T>((int)linearLength, mustPin) : (new T[linearLength]);
+            return new Tensor<T>(values, lengths.ToArray(), mustPin);
+        }
+
+        public static Tensor<T> Create<T>(bool mustPin, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanHelpers.CalculateTotalLength(lengths);
+            T[] values = mustPin ? GC.AllocateArray<T>((int)linearLength, mustPin) : (new T[linearLength]);
+            return new Tensor<T>(values, lengths.ToArray(), strides.ToArray(), mustPin);
+        }
+
+        public static Tensor<T> Create<T>(T[] values, ReadOnlySpan<nint> lengths)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanHelpers.CalculateTotalLength(lengths);
+            if (linearLength != values.Length)
+                throw new Exception();
+
+            return new Tensor<T>(values, lengths.ToArray(), false);
+        }
+
+        public static Tensor<T> Create<T>(T[] values, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanHelpers.CalculateTotalLength(lengths);
+            if (linearLength != values.Length)
+                throw new Exception();
+
+            return new Tensor<T>(values, lengths.ToArray(), strides.ToArray(), false);
+        }
+
+        //public static Tensor<T> Create(T* address, ReadOnlySpan<nint> lengths);
+        //public static Tensor<T> Create(T* address, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides);
```

Comment:
?

Reply:
Removed since for now Tensor doesn't support this.